polkadot/node/subsystem-util/src/inclusion_emulator/mod.rs

```diff
@@ -600,6 +600,18 @@ impl Fragment {
 		validation_code_hash: &ValidationCodeHash,
 		persisted_validation_data: &PersistedValidationData,
 	) -> Result<ConstraintModifications, FragmentValidityError> {
+		// Filter UMP signals and the separator.
```

Comment:
shouldn't we also have some limit for the number of ump signals?

Reply:
I see in the runtime we allow at most 2 ump signals. we should do the same here. We should actually also have a constant defined for this in the primitives module that is used between the node and runtime